sources/mysql/infoschema.go

```diff
@@ -22,99 +22,40 @@ import (
-		if !ok {
-			conv.Stats.BadRows[srcTable] += conv.Stats.Rows[srcTable]
-			conv.Unexpected(fmt.Sprintf("Can't get schemas for table %s", srcTable))
-			continue
-		}
-		srcCols := srcSchema.ColNames
-		if len(srcCols) == 0 {
-			conv.Unexpected(fmt.Sprintf("Couldn't get source columns for table %s ", t.name))
-			continue
-		}
-		colNameList := buildColNameList(srcSchema, srcCols)
-		// MySQL schema and name can be arbitrary strings.
-		// Ideally we would pass schema/name as a query parameter,
-		// but MySQL doesn't support this. So we quote it instead.
-		q := fmt.Sprintf("SELECT %s FROM `%s`.`%s`;", colNameList, t.schema, t.name)
-		rows, err := db.Query(q)
-		if err != nil {
-			conv.Unexpected(fmt.Sprintf("Couldn't get data for table %s : err = %s", t.name, err))
-			continue
-		}
-		defer rows.Close()
-		srcCols, _ = rows.Columns()
-		spTable, err := internal.GetSpannerTable(conv, srcTable)
-		if err != nil {
-			conv.Unexpected(fmt.Sprintf("Couldn't get spanner table : %s", err))
-			continue
-		}
-		spCols, err := internal.GetSpannerCols(conv, srcTable, srcCols)
-		if err != nil {
-			conv.Unexpected(fmt.Sprintf("Couldn't get spanner columns for table %s : err = %s", t.name, err))
-			continue
-		}
-		spSchema, ok := conv.SpSchema[spTable]
-		if !ok {
-			conv.Stats.BadRows[srcTable] += conv.Stats.Rows[srcTable]
-			conv.Unexpected(fmt.Sprintf("Can't get schemas for table %s", srcTable))
-			continue
-		}
-		v, scanArgs := buildVals(len(srcCols))
-		for rows.Next() {
-			// get RawBytes from data.
-			err = rows.Scan(scanArgs...)
-			if err != nil {
-				conv.Unexpected(fmt.Sprintf("Couldn't process sql data row: %s", err))
-				// Scan failed, so we don't have any data to add to bad rows.
-				conv.StatsAddBadRow(srcTable, conv.DataMode())
-				continue
-			}
-			values := valsToStrings(v)
-			ProcessDataRow(conv, srcTable, srcCols, srcSchema, spTable, spCols, spSchema, values)
-		}
+//Functions below implement the common.InfoSchema interface
+func (isi InfoSchemaImpl) GetToDdl() common.ToDdl {
+	return ToDdlImpl{}
+}
+
+func (isi InfoSchemaImpl) GetTableName(dbName string, tableName string) string {
+	return tableName
+}
+
+func (isi InfoSchemaImpl) GetRowsFromTable(conv *internal.Conv, db *sql.DB, table common.SchemaAndName) (*sql.Rows, error) {
+	srcSchema := conv.SrcSchema[table.Name]
+	srcCols := srcSchema.ColNames
+	if len(srcCols) == 0 {
+		conv.Unexpected(fmt.Sprintf("Couldn't get source columns for table %s ", table.Name))
+		return nil, nil
 	}
+	colNameList := buildColNameList(srcSchema, srcCols)
+	// MySQL schema and name can be arbitrary strings.
+	// Ideally we would pass schema/name as a query parameter,
+	// but MySQL doesn't support this. So we quote it instead.
+	q := fmt.Sprintf("SELECT %s FROM `%s`.`%s`;", colNameList, table.Schema, table.Name)
+	rows, err := db.Query(q)
+	return rows, err
 }
 
 // Building list of column names to support mysql spatial datatypes instead of
```

Comment:
Suggest moving this comment into GetRowsFromTable (right before the call to buildColNameList). I think it makes the code easier to read there because that's where the question comes up: why don't we just use a SELECT *?

And that also solves the go function comment issue: function comments should start with the name of the function.